ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Ledger/Query.hs

```diff
@@ -309,14 +309,19 @@ answerQueryAnytime HardForkLedgerConfig{..} =
 -------------------------------------------------------------------------------}
 
 data QueryHardFork xs result where
-  GetInterpreter :: QueryHardFork xs (History.Interpreter xs)
-  GetCurrentEra  :: QueryHardFork xs (EraIndex xs)
+  GetInterpreter  :: QueryHardFork xs (History.Interpreter xs)
+  GetCurrentEra   :: QueryHardFork xs (EraIndex xs)
+  GetLedgerCfg    :: QueryHardFork xs (HardForkLedgerConfig xs)
 
 deriving instance Show (QueryHardFork xs result)
 
 instance All SingleEraBlock xs => ShowQuery (QueryHardFork xs) where
-  showResult GetInterpreter = show
-  showResult GetCurrentEra  = show
+  showResult query = case query of
+    GetInterpreter -> show
+    GetCurrentEra  -> show
+    -- As the HardForkLedgerConfig is quite large and we want to avoid excessive
+    -- debug output, we truncate the contents of HardForkLedgerConfig here.
+    GetLedgerCfg   -> const "HardForkLedgerConfig{..}"
```

Comment:
Would it be easy to use `Data.Data.showConstr` instead of hardcoding the current ctor name?

Reply:
Good idea!

Reply:
Humm looks like I'll need to add a `Data (HardforkLedgerConfig xs)` constraint here and in various other places. GHC can't find an instance because we are polymorphic in `xs`. This doesn't quite seem worth it. Is there some other way to do this that I'm missing? Using `Typeable` to get the type name has a similar problem. A simpler solution could be to just add an explicit patter match so that if the constructor name changes, we'll get a compiler error any way:
```haskell
GetLedgerCfg   -> \HardForkLedgerConfig{} -> "HardForkLedgerConfig{..}"
```

Reply:
Yeah, that's a good compromise 👍